components/org.wso2.carbon.event.outbound.adapter.websubhub/src/main/java/org/wso2/carbon/event/outbound/adapter/websubhub/util/WebSubHubEventAdapterUtil.java

```diff
@@ -0,0 +1,212 @@
+     *
+     * @param eventPayload Event payload object.
+     * @param eventUri     Event URI.
+     * @param topic        Topic name.
+     * @param tenantDomain Tenant domain.
+     * @return Security Event Token payload.
+     */
+    public static SecurityEventTokenPayload buildSecurityEventToken(EventPayload eventPayload, String eventUri,
+                                                                    String topic, String tenantDomain) {
+
+        SecurityEventTokenPayload securityEventTokenPayload = new SecurityEventTokenPayload();
+        securityEventTokenPayload.setIss(EVENT_ISSUER);
+        securityEventTokenPayload.setIat(System.currentTimeMillis());
+        securityEventTokenPayload.setJti(UUID.randomUUID().toString());
+        securityEventTokenPayload.setAud(getAudience(topic, tenantDomain));
+        Map<String, EventPayload> eventMap = new HashMap<>();
+        eventMap.put(eventUri, eventPayload);
+        securityEventTokenPayload.setEvent(eventMap);
+        return securityEventTokenPayload;
+    }
+
+    /**
+     * Get audience for the given topic and tenant.
+     *
+     * @param topic        Topic name.
+     * @param tenantDomain Tenant domain.
+     * @return Audience value.
+     */
+    private static String getAudience(String topic, String tenantDomain) {
+
+        StringBuilder audience = new StringBuilder(AUDIENCE_BASE_URL);
+        audience = audience.append(tenantDomain).append(URL_SEPARATOR).append(topic);
+        return audience.toString();
+    }
+
+    /**
+     * Publish event to the web sub hub as an asynchronous API call.
+     *
+     * @param securityEventTokenPayload Security Event Token object.
+     * @param tenantDomain              Tenant domain.
+     * @param topic                     Topic name.
+     * @param webSubHubBaseUrl          Web sub hub base url.
+     */
+    public static void makeAsyncAPICall(SecurityEventTokenPayload securityEventTokenPayload, String tenantDomain,
+                                        String topic, String webSubHubBaseUrl) {
+
+        String url = buildURL(topic, tenantDomain, webSubHubBaseUrl);
+        try {
+            HttpPost request = new HttpPost(url);
+            request.setHeader(ACCEPT, TYPE_APPLICATION_JSON);
+            request.setHeader(CONTENT_TYPE, TYPE_APPLICATION_JSON);
+            request.setHeader(CORRELATION_ID_REQUEST_HEADER, getCorrelationID());
+
+            ObjectMapper mapper = new ObjectMapper();
+            String jsonString = mapper.writeValueAsString(securityEventTokenPayload);
+
+            request.setEntity(new StringEntity(jsonString));
+            CloseableHttpAsyncClient client = WebSubHubEventAdapterDataHolder.getInstance().getClientManager()
+                    .getClient(tenantDomain);
+
+            if (log.isDebugEnabled()) {
+                log.debug("Publishing event data to WebSubHub. URL: " + url + " tenant domain: "
+                        + tenantDomain + " payload:");
```

Comment:
Incomplete log. Let's not log the entire payload as it will include PII and other sensitive information

Reply:
resolved in https://github.com/wso2-extensions/identity-outbound-adapters/pull/1/commits/c57f2891f9316bc1a7008f5b74a69923ec8e89b0